### PR TITLE
[SNMP] Add constant metric in profiles

### PR DESCRIPTION
**Pull request by TCheruy**
### What does this PR do?
This PR implements constant metrics in SNMP profiles.

Example usage:
```
  - MIB: AIRESPACE-WIRELESS-MIB
    table:
      OID: 1.3.6.1.4.1.14179.2.2.1
      name: bsnAPTable
    symbols:
      - name: accessPoints
        constant_value_one: true
    metric_tags:
      - tag: ap_name
        column:
          OID: 1.3.6.1.4.1.14179.2.2.1.1.3
          name: bsnAPName
      - tag: ap_oper_status
        column:
          OID: 1.3.6.1.4.1.14179.2.2.1.1.6
          name: bsnAPOperationStatus
        mapping:
          1: associated
          2: disassociating
          3: downloading
```

Metric `snmp.accessPoints` will be emitted with a value of 1 and tags from `metric_tags`

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/collector/corechecks/snmp/internal/report/report_metrics_test.go [hunk 1]
```diff
@@ -329,6 +334,37 @@ func Test_metricSender_reportMetrics(t *testing.T) {
...
+							Value: float64(1),
+						},
+					},
+				},
+			},
+			expectedMethod:     "Gauge",
+			expectedMetricName: "snmp.constantMetric",
+			expectedValue:      float64(1),
+			expectedTags:       []string{"status:1"},
+			expectedSubMetrics: 3,
```
Comment (AlexandreYang) [inline pkg/collector/corechecks/snmp/internal/report/report_metrics_test.go:366]: **💬 suggestion:** ‏Can we assert all 3 metrics instead?

You can possibly create a expectedMetrics struct here https://github.com/DataDog/datadog-agent/pull/16547/files#diff-a04138749aca08251bdd24d73eb71c99bbb4f12e1d8e82ddcab7f4c9bf5ec3baR315-R325

and then loop during the assertion. WDYT?

File: pkg/collector/corechecks/snmp/internal/report/report_utils.go [hunk 2]
```diff
@@ -170,3 +170,26 @@ func getInterfaceConfig(interfaceConfigs []snmpintegration.InterfaceConfig, inde
 	}
 	return snmpintegration.InterfaceConfig{}, fmt.Errorf("no matching interface found for index=%s, tags=%s", index, tags)
 }
+
+// getConstantMetricValues retrieve all metric tags indexes and set their value as 1
+func getConstantMetricValues(mtcl checkconfig.MetricTagConfigList, values *valuestore.ResultValueStore) map[string]valuestore.ResultValue {
```
Comment (AlexandreYang) [inline pkg/collector/corechecks/snmp/internal/report/report_utils.go:175]: **💬 suggestion:** ‏Could you add a unit test for this function? 

Small functions like this one should be ideally unit-tested, and ideally for all code paths (happy path and error paths e.g. `error getting column value`).

**Review by AlexandreYang — CHANGES_REQUESTED** [review]
Nice. Looks good overall, few small comments.

File: pkg/collector/corechecks/snmp/internal/checkconfig/config_validate_enrich.go [hunk 3]
```diff
@@ -117,13 +126,17 @@ func validateEnrichMetadata(metadata MetadataConfig) []string {
 	return errors
 }
 
-func validateEnrichSymbol(symbol *SymbolConfig) []string {
+func validateEnrichSymbol(symbol *SymbolConfig, symbolContext SymbolContext) []string {
```
[the remark this answers is not in the document]
Comment (TCheruy) [reply pkg/collector/corechecks/snmp/internal/checkconfig/config_validate_enrich.go:129]: Of course good catch thank you!

File: pkg/collector/corechecks/snmp/internal/report/report_utils.go (hunk 2 above)
Comment (TCheruy) [reply pkg/collector/corechecks/snmp/internal/report/report_utils.go:175]: Of course thank you!  Test got lost in my local history 🤦‍♂️

**Review by hestonhoffman — APPROVED** [review]

File: pkg/collector/corechecks/snmp/internal/report/report_metrics.go [hunk 4]
```diff
@@ -122,9 +122,16 @@ func (ms *MetricSender) reportColumnMetrics(metricConfig checkconfig.MetricsConf
...
-		metricValues, err := getColumnValueFromSymbol(values, symbol)
-		if err != nil {
-			continue
+		var metricValues map[string]valuestore.ResultValue
+
+		if symbol.ConstantValueOne {
+			metricValues = getConstantMetricValues(metricConfig.MetricTags, values)
+		} else {
+			var err error
+			metricValues, err = getColumnValueFromSymbol(values, symbol)
```
Comment (AlexandreYang) [inline pkg/collector/corechecks/snmp/internal/report/report_metrics.go:131]:
```suggestion
			metricValues, err := getColumnValueFromSymbol(values, symbol)
```
🥜 shorter syntax?

File: pkg/collector/corechecks/snmp/internal/report/report_metrics.go [hunk 5]
```diff
@@ -122,9 +122,16 @@ func (ms *MetricSender) reportColumnMetrics(metricConfig checkconfig.MetricsConf
...
+			if err != nil {
+				continue
```
Comment (AlexandreYang) [inline pkg/collector/corechecks/snmp/internal/report/report_metrics.go:133]: 🥜 I know it wasn't there before, but maybe we can make a debug log here.

File: pkg/collector/corechecks/snmp/internal/report/report_utils_test.go [hunk 6]
```diff
@@ -903,3 +903,206 @@ func Test_getInterfaceConfig(t *testing.T) {
 		})
 	}
 }
+
+func Test_getContantMetricValues(t *testing.T) {
```
Comment (AlexandreYang) [inline pkg/collector/corechecks/snmp/internal/report/report_utils_test.go:907]: thanks 👏

File: pkg/collector/corechecks/snmp/internal/report/report_metrics.go (hunk 4 above)
Comment (TCheruy) [reply pkg/collector/corechecks/snmp/internal/report/report_metrics.go:131]: `metricValues` is declared line 125 (outside of the if statement) so using `:=` gives `Unused variable 'metricValues'`, I don't know how to better write this 🙁

**Review by AlexandreYang — APPROVED** [review]

File: pkg/collector/corechecks/snmp/internal/report/report_metrics.go (hunk 4 above)
Comment (AlexandreYang) [reply pkg/collector/corechecks/snmp/internal/report/report_metrics.go:131]: ah yes, it's fine then :)

**Review by AlexandreYang — APPROVED** [review]

**Pull request merged**